### DISCUSSION TITLE
Merge several YAML files and extract one value from the rightmost one

### DISCUSSION THREAD

**Discussion by victor-sudakov** [mikefarah/yq#1943] [Q&A] [answered]
Colleagues, 
I cannot wrap around my head how to perform a very simple task: merge several YAML files and take the value from the last one (if it is present there), or the last but one if not present in the last one. E.g. `yq '.image.phptests.tag' values.yaml values-local.yaml` should output the value of the tag from the rightmost file where it is present. In other words, what Helm does when given multiple override `--values` files.
Can you help me formulate the query?

**Comment by mikefarah** [comment]
You can use the multiply `*`  operator which merges objects together, and then lookup your property:

```
yq -n 'load("values.yaml") * load("values-local.yaml")  | .image.phptests.tag' 
```

See https://mikefarah.gitbook.io/yq/operators/multiply-merge for different merge examples and options

**Reply by victor-sudakov** [reply, marked as answer]
Hello @mikefarah Can I supply the files to merge as command line parameters because 1) there maybe more than two and their number may vary and 2) their names can be variables and I'd avoid placing variables into single ticks?

Do I have to use the syntax `yq eval-all '. as $item ireduce ({}; . * $item ) | .image.phptests.tag' values*.yml` in this case?

**Reply by mikefarah** [reply]
Yep that's right 👍🏼

**Reply by victor-sudakov** [reply]
Thank you @mikefarah !
